enterprise/backend/src/metabase_enterprise/semantic_search/index.clj

```diff
@@ -82,54 +81,60 @@
+                                     (jdbc/execute! connectable (records->sql db-records))
+                                     ;; TODO should this return (or at least log) the number of docs actually
+                                     ;; updated, not just the number in the batch?
+                                     (u/prog1 (->> db-records (map :model) frequencies)
+                                       (log/trace "semantic search processed a batch of" (count db-records)
+                                                  "documents with frequencies" <>)))))
+                        (partial merge-with +)
+                        (map vector documents embeddings))
+      (log/trace "semantic search processed" (count documents) "total documents with frequencies" <>))))
 
 (defn- batch-delete-ids!
-  ([model ids->sql ids]
-   (batch-delete-ids! @db/data-source model ids->sql ids))
-  ([connectable model ids->sql ids]
-   (when (seq ids)
-     (u/prog1 (->> (transduce (comp (partition-all *batch-size*)
-                                    (map (fn [ids]
-                                           (jdbc/execute! connectable (ids->sql ids))
-                                           (u/prog1 (count ids)
-                                             (log/trace "semantic search deleted a batch of" <>
-                                                        "documents with model type" model)))))
-                              +
-                              ids)
-                   (array-map model))
-       (log/trace "semantic search deleted" (get <> model) "total documents with model type" model)))))
+  [connectable model ids->sql ids]
+  (when (seq ids)
+    (u/prog1 (->> (transduce (comp (partition-all *batch-size*)
+                                   (map (fn [ids]
+                                          (jdbc/execute! connectable (ids->sql ids))
+                                          (u/prog1 (count ids)
+                                            (log/trace "semantic search deleted a batch of" <>
+                                                       "documents with model type" model)))))
+                             +
+                             ids)
+                  (array-map model))
+      (log/trace "semantic search deleted" <> "total documents with model type" model))))
 
 (defn- db-records->update-set
   [db-records]
   (let [update-keys (-> db-records first (dissoc :id :model :model_id) keys)
         excluded-kw (fn [column] (keyword (str "excluded." (name column))))]
     (zipmap update-keys (map excluded-kw update-keys))))
 
+(defn default-index
+  "Returns the default index spec for a model."
+  [embedding-model]
+  (let [{:keys [model-name provider vector-dimensions]} embedding-model
+        table-name (str "index_table__" provider "__" model-name "__" vector-dimensions)]
+    {:embedding-model embedding-model
+     :table-name table-name
+     :index-name (str table-name "__hnsw_idx")}))
```

Comment:
Nice, this is effectively the same fix as #61312. I will close that PR in favor of this one.